lms/djangoapps/grades/tasks.py

```diff
@@ -115,6 +115,16 @@ def recalculate_subsection_grade_v3(self, **kwargs):
     _recalculate_subsection_grade(self, **kwargs)
 
 
+@task()
+def add(x, y):
+    return x + y
+
+
+@task
+def tsum(x):
+    return sum(x)
```

Comment:
Sample execution (local):

```
edxapp@vagrant:~/edx-platform$ ./manage.py lms --settings=devstack_with_worker shell
Python 2.7.12 (default, Nov 19 2016, 06:48:10) 
[GCC 5.4.0 20160609] on linux2
Type "help", "copyright", "credits" or "license" for more information.
(InteractiveConsole)
>>> from lms.djangoapps.grades.tasks import add, tsum; from celery import chord
>>> c = chord(add.s(i, i) for i in xrange(10))(tsum.s())
```

Worker shell:
```
2017-04-19 20:25:13,072 INFO 4237 [celery.worker.strategy] strategy.py:55 - Received task: lms.djangoapps.grades.tasks.add[12dd23da-f759-49ab-9436-a5d14f58f382]
2017-04-19 20:25:13,076 INFO 4237 [celery.worker.job] job.py:401 - Task lms.djangoapps.grades.tasks.add[12dd23da-f759-49ab-9436-a5d14f58f382] succeeded in 0.00309259700589s: 0
2017-04-19 20:25:13,077 INFO 4237 [celery.worker.strategy] strategy.py:55 - Received task: lms.djangoapps.grades.tasks.add[2f9dd04a-09d6-4a93-8f3f-eecf4cd0ec94]
2017-04-19 20:25:13,079 INFO 4237 [celery.worker.job] job.py:401 - Task lms.djangoapps.grades.tasks.add[2f9dd04a-09d6-4a93-8f3f-eecf4cd0ec94] succeeded in 0.00124310200044s: 2
2017-04-19 20:25:13,083 INFO 4237 [celery.worker.strategy] strategy.py:55 - Received task: lms.djangoapps.grades.tasks.add[d149d4d5-1169-4271-8187-b67ce99b389d]
2017-04-19 20:25:13,084 INFO 4237 [celery.worker.strategy] strategy.py:55 - Received task: lms.djangoapps.grades.tasks.add[b0ef674c-96c3-47a2-86bd-4993f50082f8]
2017-04-19 20:25:13,085 INFO 4237 [celery.worker.job] job.py:401 - Task lms.djangoapps.grades.tasks.add[d149d4d5-1169-4271-8187-b67ce99b389d] succeeded in 0.00110853801016s: 4
2017-04-19 20:25:13,086 INFO 4237 [celery.worker.job] job.py:401 - Task lms.djangoapps.grades.tasks.add[b0ef674c-96c3-47a2-86bd-4993f50082f8] succeeded in 0.00148267600161s: 6
2017-04-19 20:25:13,087 INFO 4237 [celery.worker.strategy] strategy.py:55 - Received task: lms.djangoapps.grades.tasks.add[410daee0-e0ff-48e9-aba3-e904e6c4861d]
2017-04-19 20:25:13,087 INFO 4237 [celery.worker.strategy] strategy.py:55 - Received task: lms.djangoapps.grades.tasks.add[7b6a0a86-f80b-4337-a031-fadb72d80600]
2017-04-19 20:25:13,090 INFO 4237 [celery.worker.job] job.py:401 - Task lms.djangoapps.grades.tasks.add[7b6a0a86-f80b-4337-a031-fadb72d80600] succeeded in 0.00150735198986s: 10
2017-04-19 20:25:13,090 INFO 4237 [celery.worker.job] job.py:401 - Task lms.djangoapps.grades.tasks.add[410daee0-e0ff-48e9-aba3-e904e6c4861d] succeeded in 0.00192271500418s: 8
2017-04-19 20:25:13,091 INFO 4237 [celery.worker.strategy] strategy.py:55 - Received task: lms.djangoapps.grades.tasks.add[0a46ace4-332c-4b73-b831-8d3f427c3c39]
2017-04-19 20:25:13,092 INFO 4237 [celery.worker.strategy] strategy.py:55 - Received task: lms.djangoapps.grades.tasks.add[63ca558a-bf6a-4b01-b6e5-2b5494a5dd6f]
2017-04-19 20:25:13,095 INFO 4237 [celery.worker.job] job.py:401 - Task lms.djangoapps.grades.tasks.add[0a46ace4-332c-4b73-b831-8d3f427c3c39] succeeded in 0.00152764499944s: 12
2017-04-19 20:25:13,096 INFO 4237 [celery.worker.job] job.py:401 - Task lms.djangoapps.grades.tasks.add[63ca558a-bf6a-4b01-b6e5-2b5494a5dd6f] succeeded in 0.00201725700754s: 14
2017-04-19 20:25:13,099 INFO 4237 [celery.worker.strategy] strategy.py:55 - Received task: lms.djangoapps.grades.tasks.add[7aac3aa3-c195-4e44-8db4-68fc2db496fc]
2017-04-19 20:25:13,105 INFO 4237 [celery.worker.strategy] strategy.py:55 - Received task: lms.djangoapps.grades.tasks.add[431a8b3b-7997-4813-98fe-e226ac72fcce]
2017-04-19 20:25:13,108 INFO 4237 [celery.worker.job] job.py:401 - Task lms.djangoapps.grades.tasks.add[7aac3aa3-c195-4e44-8db4-68fc2db496fc] succeeded in 0.00141846000042s: 16
2017-04-19 20:25:13,623 INFO 4237 [celery.worker.strategy] strategy.py:55 - Received task: lms.djangoapps.grades.tasks.tsum[4b3769ef-8085-4847-b779-1a415a961a69]
2017-04-19 20:25:13,626 INFO 4237 [celery.worker.job] job.py:401 - Task lms.djangoapps.grades.tasks.tsum[4b3769ef-8085-4847-b779-1a415a961a69] succeeded in 0.00122585600184s: 90
2017-04-19 20:25:13,627 INFO 4237 [celery.worker.job] job.py:401 - Task lms.djangoapps.grades.tasks.add[431a8b3b-7997-4813-98fe-e226ac72fcce] succeeded in 0.520675369s: 18
```

Notice that the last sub-task of the group takes a long time to resolve - I'm assuming that it calls the after-chord callback, and its "execution time" includes all that overhead.